x-pack/auditbeat/module/system/login/utmp.go

```diff
@@ -0,0 +1,571 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux,cgo
+
+package login
+
+// #include <utmp.h>
+// #include <stdlib.h>
+import "C"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/user"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/auditbeat/datastore"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+const (
+	bucketKeyFileRecords   = "file_records"
+	bucketKeyLoginSessions = "login_sessions"
+)
+
+// Inode represents a file's inode on Linux.
+type Inode uint64
+
+// FileRecord represents a UTMP file at a point in time.
+type FileRecord struct {
+	Inode    Inode
```

Comment:
In filebeat we also store the device id.

Reply:
I think for now it's unlikely that UTMP files would be stored on separate file systems. I.e. it would be mean different parts of `/var/log` are on different devices. I'm not even sure that's possible.